tests/unit/yaml/dummy_gateway.py

```diff
@@ -63,3 +75,44 @@ async def run_server(self):
     async def shutdown(self):
         self.server.should_exit = True
         await self.server.shutdown()
+
+
+class FlaskDummyGateway(Gateway):
+    async def setup_server(self):
+        # step 1: create an app and define the service endpoint
+        app = Flask(__name__)
```

Comment:
also we need to check how to properly terminate this server (look at `should_exit`)